common/database_handler.py

```diff
@@ -71,5 +86,14 @@ def __init__(self):
     def __enter__(self):
         return self.conn
 
-    def __exit__(self, *_):
+    def __exit__(self, exception_type, exception_value, traceback):
         DatabasePoolHandler.return_connection(self.conn)
+        if exception_type is InterfaceError or exception_type is DatabaseError:
+            LOGGER.exception(f"Exception type: {exception_type}. "
```

Comment:
please use LOGGER.error instead of LOGGER.exception here, LOGGER.exception catches and prints currently active exception

try to use `LOGGER.error("Exception type: %s", exception_type)` style of formatting to align with project style